contribute/design/1.x/cstor/pool/pool_expansion_by_expanding_disk.md

```diff
@@ -0,0 +1,134 @@
+
+Below is high-level info to explain the Day2 operations that are currently supported.
+
+- Pool Expansion When BlockDevice/raid group added:
+	cstor-pool-manager will iterate over all the BlockDevices that exist in CStorPoolInstance resource and verify whether the corresponding BlockDevice/raid group is part of the pool. If that BlockDevice/raid group is not part of the pool then pool expansion will be triggered on newly added BlockDevice/raid group.
+
+- BlockDevice Replacement In Pool:
+    cstor-pool manager will iterate over all the BlockDevices present in CstorPoolInstance and verifies whether the current blockdevice is for BlockDevice replacement request[How cstor-pool-manager can know? We are maintaining annotation called openebs.io/predecessor: <replaced_blockdevice_name> on the claim of that blockdevice which will be populated by admission server on replacement request]? If requested BlockDevice is valid for replacement then the pool manager will perform BlockDevice replacement operation on the pool.
+
+### Proposed Implementations
+
+- CStor-pool Manager inbuilt CSPI controller watches for CSPI resources and tries to get the desired state via every sync operation. As part of reconciliation logic, it will iterate over all the BlockDevices that exist on CStorPoolInstance resource if it found BlockDevice resource capacity(considerable changes i.e difference is > 500MB) is greater than the existing capacity of BlockDevice on CstorPoolInstanceBlockDevice then we will follow below-mentioned steps to perform the [pool expansion](#Pool_Expansion_Steps).
+- [What if cStor is consuming part of partitioned disk/entire partitioned disk? The User has to extend the partition manually after that NDM will update the capacity on the corresponding blockdevice CR. Once NDM updates the capacity on BlockDevice in next reconciliation of CSPI Pool Expansion will be triggered].
+- [What if only one blockdevice got expanded in case of raid/mirror configuration? Pool expansion operation will be triggered but there won't be any change in the size of the pool].
+
+NOTE: During the cStor pool creation time and pool import time `autoexpand` property will be configured to `on`(configuring during import time will support for pools which were migrated from SPC to CSPC).
```

Comment:
can you describe why we need this property to be set to on?

Reply:
I followed the [doc](https://github.com/openebs/openebs-docs/blob/day_2_ops/docs/resize-single-disk-pool.md#3-set-your-zpool-with-autoexpand-on-by-default-it-is-set-to-off) and assumed that we need to set it for expansion since we are at user space of ZFS(autoexpand configuration will not help) we no need to set the autoexpand property. We are informing to zrepl via `zpool online -e <pool_name> <disk_path>` command I will remove setting this property.

Reply:
autoexpand is needed when we are replacing the disk to higher size and want pool to be expanded automatically. Add a section where we expand the pool by replacing the disk with higher size.

Reply:
As of no relevant place for this, I will add as a note under the above point.

Reply:
I will update.